migration/migrator/migrations/course/20231206173634_course_materials.py

```diff
@@ -0,0 +1,40 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    
+    # Add columns to the course_materials table.
+    database.execute("""
+                     ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS uploaded_by character varying(255) REFERENCES users(user_id) DEFAULT NULL;
+                     ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS upload_date TIMESTAMP WITH TIME ZONE DEFAULT NULL;
+                     ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS last_edit_by character varying(255) REFERENCES users(user_id) DEFAULT NULL;
+                     ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS last_edit_date TIMESTAMP WITH TIME ZONE DEFAULT NULL;
+                     UPDATE course_materials SET last_edit_by = uploaded_by, last_edit_date = upload_date;
```

Comment:
This `UPDATE` command will still overwrite any existing data if the table already exists.  You need to add a `WHERE` clause to only update if the field is null.  Also, should the new columns be non-null, given that you're essentially setting them to default values here?

Reply:
In previous commits, I mistakenly redid the migration function because the `UPDATE` query was still present. However, we do not need that update query.

The implementation we are aiming for is that during the migration of the database, the default value for new columns, for which some data already exists, will be set to `NULL`. 

When new course material is added, the upload-related column will be populated with the data. Additionally, when the course is edited, the last_edit related column will be populated. This is the desired behavior from the website.

In the latest commit, all of these tasks are now performed.